janusgraph-es/src/main/java/org/janusgraph/diskstorage/es/ElasticSearchIndex.java

```diff
@@ -243,6 +244,12 @@
         new ConfigOption<>(ELASTICSEARCH_NS, "retry_on_conflict",
             "Specify how many times should the operation be retried when a conflict occurs.", ConfigOption.Type.MASKABLE, 0);
 
+    public static final ConfigOption<Boolean> ENABLE_INDEX_STORE_NAMES_CACHE =
```

Comment:
I think we don't need addition option for that

Simply we may use cache with 10k LRU enties

Reply:
Did you check performance of 10k LRU cache and concurrentHashMap? I checked that it is faster to get a name from map than making contact for 50k and less entities. What if user has more indexes (unlikely but possible)? In such scenario the will be forced to use cache that is slower than string concatenation.

Reply:
Also, LRU cache may evict some names which we would like to avoid because it is a permanent cache

Reply:
Yes you right concurrenthashmap mush faster than cache. But is so rare case when user would be change this option by memory usages cause, i think

Reply:
You are right. Most likely nobody will use this option and most likely it will be `true` for all usecases but I think, it would be good to have this option in case the user has very unusual usecase. 

Also, this optimization was introduced here #1695 . This optimization is really tiny, so, regular user won't see any difference but still it speedups index store name retrieval for about 3-4 times faster for regular usages and reduces memory consumptions for unnecessary string concatenations.